framework/infrastructure/traffic_director.py

```diff
@@ -171,13 +176,19 @@ def setup_routing_rule_map_for_grpc(self, service_host, service_port):
         self.create_target_proxy()
         self.create_forwarding_rule(service_port)
 
+        if self.enable_dualstack:
+            self.create_target_proxy_ipv6()
+            self.create_forwarding_rule_ipv6(service_port)
+
     def cleanup(self, *, force=False):
         # Cleanup in the reverse order of creation
         self.delete_firewall_rules(force=force)
         self.delete_forwarding_rule(force=force)
         self.delete_alternative_forwarding_rule(force=force)
         self.delete_target_http_proxy(force=force)
         self.delete_target_grpc_proxy(force=force)
+        if self.enable_dualstack:
+            self.delete_target_proxy_ipv6(force=force)
```

Comment:
We need to clean up forwarding rule ipv6 too.

Reply:
Had both in delete_forwarding_rule but moved ipv6 to separate method to keep consistent with convention

Reply:
Forwarding rule should be deleted before target proxy, or it'll fail because of the dependency order